File: c++/simple_calculator.cpp
// Day 1: Simple Calculator

// Concepts implemented:
//   -> 'if/else' conditions
//   -> 'while' loop
//   -> functions


#include<iostream>

//calculator functions
int add(int num1, int num2)
{
	return num1 + num2;
}
int subtract(int num1, int num2)
{
	return num1 - num2;
}
int multiply(int num1, int num2)
{
	return num1 * num2;
}
int quotient(int num1, int num2)
{
	return num1 / num2;
}
int remainder(int num1, int num2)
{
	return num1 % num2;
}

//print functions
void menu()
{
	std::cout << "*********** MENU ***********\n\n";
	std::cout << "KEYCODE          FUNCTION\n";
	std::cout << "  0                Exit\n";
	std::cout << "  1                Add\n";
	std::cout << "  2              Subtract\n";
	std::cout << "  3              Multiply\n";
	std::cout << "  4               Divide\n";
	std::cout << "\n*****************************\n\n";
}
void exitMessage() 
{
	std::cout << "\n\nBye bye! See you next time! :)\n\n";
}


int main()
{
	int userInput = -1;
	bool isExited = false; //check if user has exited the calculator
	int num1 = -1;
	int num2 = -1;

	std::cout << "Welcome to my Simple Calculator! I got the solution to all your arithmetic needs!\n\n";

	while (!isExited)
	{
		menu(); //print menu

		do
		{
			std::cout << "Please enter the function you would like to perform: ";
			std::cin >> userInput;

		} while (userInput < 0 || userInput > 4); //make sure it's between 0 and 4

		if (userInput == 0) //Exit
		{
			isExited = true;
			continue;
		}
		else if (userInput == 1) //Addition
		{
			std::cout << "\n------ ADDITION ------\n";
			std::cout << "FORMAT: A + B\n\n";

			std::cout << "Enter A: ";
			std::cin >> num1;

			std::cout << "Enter B: ";
			std::cin >> num2;
			
			std::cout << "\n-> " << num1 << " + " << num2 << " = " << add(num1, num2);
			
			std::cout << "\n----------------------\n\n\n\n";
		}
		else if (userInput == 2) //Subtraction
		{
			std::cout << "\n------ SUBTRACTION ------\n";

			std::cout << "FORMAT: A - B\n\n";

			std::cout << "Enter A: ";
			std::cin >> num1;

			std::cout << "Enter B: ";
			std::cin >> num2;

			std::cout << "\n ->" << num1 << " - " << num2 << " = " << subtract(num1, num2);

			std::cout << "\n----------------------\n\n\n\n";
		}
		else if (userInput == 3) //Multiplication
		{
			std::cout << "\n------ MULTIPLICATION ------\n";

			std::cout << "FORMAT: A * B\n\n";

			std::cout << "Enter A: ";
			std::cin >> num1;

			std::cout << "Enter B: ";
			std::cin >> num2;

			std::cout << "\n-> " << num1 << " * " << num2 << " = " << multiply(num1, num2);

			std::cout << "\n----------------------\n\n\n\n";
		}
		else if (userInput == 4) //Division
		{
			std::cout << "\n------ DIVISION ------\n";

			std::cout << "FORMAT: A / B\n\n";

			std::cout << "Enter A: ";
			std::cin >> num1;

			std::cout << "Enter B: ";
			std::cin >> num2;

			std::cout << "\n-> " << num1 << " / " << num2 << " = " << quotient(num1, num2) << " r " << remainder(num1, num2);

			std::cout << "\n----------------------\n\n\n\n";
		}

	}
	
	exitMessage();

	return 0;
}

File: python/simple_calculator.py
# Day 1: Simple Calculator

# Concepts Implemented:
#    - Functions
#    - Loops ('while')
#    - Conditional statements ('if-else')

#Thoughts: Python is so much simpler compared to C++ :')

# --------------------------------------
# CALCULATOR FUNCTIONS:
def add(num1, num2):
    return num1 + num2

def subtract(num1, num2):
    return num1 - num2

def multiply(num1, num2):
    return num1 * num2

def divide(num1, num2):
    return num1/num2


# ---------------------------------------
# PRINT FUNCTIONS
def printMenu():
    print("************ MENU ************")
    print("KEYCODE          FUNCTION")
    print("  0                Exit")
    print("  1              Addition")
    print("  2             Subtraction")
    print("  3            Multiplication")
    print("  4               Division")
    print("******************************")
 
 
def printExitMessage():
    print("\nThat's it for now! See you next time! :)\n\n")



# -------------------------------------
# MAIN:
exited = False
A = -1
B = -1

print("Welcome to my calculator! :D \n\n")

while not exited:
    
    printMenu()
    
    #simulating a do-while loop for user input
    while True:
        userInput = int(input("Please enter the function you would like to perform: "))
        
        if userInput <= 4 and userInput >= 0: #input is in valid range
            break
        
        print("\nERROR: NUMBER OUT OF RANGE!\n")
    
    #user input completed
    
    if userInput == 0: #exit
        exited = True
        continue
    
    elif userInput == 1: #addition
        print("\n-------- ADDITION --------")
        print("FORMAT: A + B")
        A = int(input("Please enter A: "))
        B = int(input("Please enter B: "))
        print(" -> ", A, "+", B, "=", add(A, B))
        print("\n--------------------------\n\n")
        
        
    elif userInput == 2: #subtraction
        print("\n-------- SUBTRACTION --------")
        print("FORMAT: A - B")
        A = int(input("Please enter A: "))
        B = int(input("Please enter B: "))
        print(" -> ", A, "-", B, "=", subtract(A, B),"\n\n")
        print("\n-----------------------------\n\n")
        
        
    elif userInput == 3: #multiplication
        print("\n-------- MULTIPLICATION --------")
        print("FORMAT: A * B")
        A = int(input("Please enter A: "))
        B = int(input("Please enter B: "))
        print(" -> ", A, "+", B, "=", multiply(A, B))
        print("\n--------------------------------\n\n")
        
        
    elif userInput == 4: #division
        print("\n-------- DIVISION --------")
        print("FORMAT: A / B")
        A = int(input("Please enter A: "))
        B = int(input("Please enter B: "))
        print(" -> ", A, "/", B, "=", divide(A, B))
        print("\n--------------------------\n\n")

    
#calculator exited
printExitMessage()




